### DISCUSSION TITLE
Can I use StorageTexture bindings with `AsBindGroup`?

### DISCUSSION THREAD

**Discussion by Droggelbecher** [bevyengine/bevy#8360] [Q&A] [answered]
I want to implement a custom material shader that uses an uint16 storage texture. Its certainly doable with implementing a custom pipeline but I wonder whether that is overkill and there is some way to do it with `#[derive(AsBindGroup)]`, but neither `#[texture(..)]` nor `#[storage(..)]` seem to lead to the right datatype?
Am I just using these wrong or is there some other trick with which I can avoid a full blown custom pipeline?

**Comment by nicopap** [comment, marked as answer]
This is something I wanted for a while, but I don't think is possible yet, see <https://github.com/bevyengine/bevy/issues/7640>. This is a limitation of the `AsBindGroup` derive macro, though there isn't anything fundamentally preventing it.
